src/compiler/checker.ts

```diff
@@ -14650,17 +14650,17 @@ namespace ts {
             return false;
         }
 
-        function addTypeToUnion(typeSet: Type[], includes: TypeFlags, type: Type) {
+        function addTypeToUnion(typeSet: Type[], includes: TypeFlags, type: Type, ignoreStrictNullChecks?: boolean) {

```

Comment:
I don't think I like this boolean, but it should be named `forceStrictNullChecks` if there's no better way to fix this bug.